code/game/objects/items/devices/transfer_valve.dm

```diff
@@ -128,22 +128,24 @@
 			toggle = 1
 
 /obj/item/transfer_valve/on_update_icon()
-	overlays.Cut()
+	. = ..()
 	underlays.Cut()
 
 	if(!tank_one && !tank_two && !attached_device)
 		icon_state = "valve_1"
 		return
 	icon_state = "valve"
 
+	var/list/cur_overlays
 	if(tank_one)
-		overlays += "[tank_one.icon_state]"
+		LAZYADD(cur_overlays, new /mutable_appearance(tank_one))
```

Comment:
We have a `mutable_appearance()` proc similar to `overlay_image()` for this.

Reply:
The proc doesn't accept an object though.